gems/smithy/spec/interfaces/client/shapes_spec.rb

```diff
@@ -20,81 +19,175 @@
+      expect(subject::Short).to be_a(shapes_module::IntegerShape)
+    end
 
-            if (expected_traits = shape['member']['traits'])
-              expect(generated_shape.member.traits).to include(expected_traits)
-            end
-          end
-        end
+    it 'timestamp is an instance of TimestampShape' do
+      expect(subject::Timestamp).to be_a(shapes_module::TimestampShape)
+    end
 
-        if shape['key'] && shape['value']
-          it 'has key and value members' do
-            expect(generated_shape.key.name).to eq('key')
-            expect(generated_shape.value.name).to eq('value')
-            expect(generated_shape.key.shape.id).to eq(shape['key']['target'])
-            expect(generated_shape.value.shape.id).to eq(shape['value']['target'])
+    context 'enum' do
+      let(:expected_member) do
+        fixture['shapes'].select { |_, s| s['type'] == 'enum' }
+                         .values
+                         .first['members']['FOO']
+      end
 
-            if (expected_traits = shape['key']['traits'])
-              expect(generated_shape.key.traits).to include(expected_traits)
-            end
+      it 'is an instance of EnumShape' do
+        expect(subject::Enum).to be_a(shapes_module::EnumShape)
+      end
 
-            if (expected_traits = shape['value']['traits'])
-              expect(generated_shape.value.traits).to include(expected_traits)
-            end
-          end
-        end
+      it 'has members' do
+        expect(subject::Enum.members.keys).to eq(%i[foo])
+        expect(subject::Enum.members[:foo]).to be_a(shapes_module::MemberShape)
+        expect(subject::Enum.members[:foo].shape).to be_a(shapes_module::StructureShape)
+        expect(subject::Enum.members[:foo].name).to eq('foo')
+        expect(subject::Enum.members[:foo].traits).to eq(expected_member['traits'])
+        expect(subject::Enum.members[:foo].shape.id).to eq(expected_member['target'])
+      end
+    end
+
+    context 'intEnum' do
+      let(:expected_member) do
+        fixture['shapes'].select { |_, s| s['type'] == 'intEnum' }
+                         .values
+                         .first['members']['BAZ']
+      end
+
+      it 'is an instance of IntEnumShape' do
+        expect(subject::IntEnum).to be_a(shapes_module::IntEnumShape)
+      end
+
+      it 'has members' do
+        expect(subject::IntEnum.members.keys).to eq(%i[baz])
+        expect(subject::IntEnum.members[:baz]).to be_a(shapes_module::MemberShape)
+        expect(subject::IntEnum.members[:baz].shape).to be_a(shapes_module::StructureShape)
+        expect(subject::IntEnum.members[:baz].name).to eq('baz')
+        expect(subject::IntEnum.members[:baz].traits).to eq(expected_member['traits'])
+        expect(subject::IntEnum.members[:baz].shape.id).to eq(expected_member['target'])
+      end
+    end
+
+    context 'list' do
+      let(:expected_member) do
+        fixture['shapes'].select { |_, s| s['type'] == 'list' }
```

Comment:
Please fix for all others (thought that went without saying..)

Reply:
My bad - missed that, will add now